### PR TITLE
1.0.0 proposal

### PR DESCRIPTION
**Pull request by Lauriethefish**
Current changes:
- Allow omitting `packageId`, which indicates that the mod works for any package.

Please give any further ideas for schema changes for 1.0.0, bearing in mind that old mods need to still be compatible.

### CODE REVIEW

File: QuestPatcher.QMod/QModManifest.cs [hunk 1]
```diff
@@ -137,10 +137,11 @@ public string VersionString
...
-            "0.1.2"
+            "0.1.2",
+            "1.0.0"
         }.ToHashSet();
 
-        private const string LatestSchemaVersion = "0.1.2";
+        private const string LatestSchemaVersion = "1.0.0";
 
         private static readonly JsonSerializerOptions SerializerOptions = new JsonSerializerOptions()
         {
```
Comment (sc2ad) [inline QuestPatcher.QMod/QModManifest.cs:150]: In this file, consider resolving the potential null references to the `QModManifest` type before merging this featureset.

**Review by sc2ad — CHANGES_REQUESTED** [review]
I think this is a reasonable change, if you wish to continue on this track, you could potentially optionally omit `packageVersion` and expect the package to work on any version of any package (should certainly be omitted for cases where `packageId` is omitted anyways).

At the very least, specifying `packageVersion` as required is redundant if `packageId` is omitted, so that should be changed, imo.

**Comment by Lauriethefish** [comment]
`packageVersion` is now optional, even if `packageId` is specified.

File: QuestPatcher.QMod/QModManifest.cs [hunk 2]
```diff
@@ -68,13 +68,16 @@ public string VersionString
...
         /// The package ID of the app that the mod is designed for.
+        /// If null, this means that the mod works for any app.
         /// </summary>
-        public string PackageId { get; set; }
+        public string? PackageId { get; set; }
 
         /// <summary>
         /// The version of the app that the mod is designed for.
+        /// If null, this means that the mod doesn't depend on a particular version of an app.
+        /// Specifying this is redundant if <see cref="PackageId"/> is specified.
```
Comment (sc2ad) [inline QuestPatcher.QMod/QModManifest.cs]: What's the verbage meant to mean here? This makes it seem like specifying the `packageVersion` is redundant if the `packageId` is specified. Isn't it the opposite that is the case?

Comment (sc2ad) [reply QuestPatcher.QMod/QModManifest.cs]: Perhaps using a word such as "omitting this is redundant..." or "specifying this is redundant... is not specified"

File: QuestPatcher.QMod/Resources/qmod.schema.json [hunk 3]
```diff
@@ -138,7 +137,7 @@
     "packageVersion": {
       "$id": "#/properties/packageVersion",
       "default": "",
-      "description": "The Android package version of the target package.",
+      "description": "The Android package version of the target package. If not specified, the mod does not depend on any particular package version. Specifying this is redundant if packageId is not specified.",
```
Comment (sc2ad) [inline QuestPatcher.QMod/Resources/qmod.schema.json:140]: ditto previous

**Review by sc2ad — APPROVED** [review]
Overall approved

**Pull request merged**